File: 11二进制求和.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string s;
        int lena = a.length()-1;
        int lenb = b.length()-1;
        int temp = 0;
        char c;
        while(lena>=0 && lenb>=0){
            c = (a[lena]-'0') + (b[lenb] - '0') +temp +'0';
            temp = 0;
            if ((c-'0')>=2)
            {
            	temp = 1;
            	c = c -2;
            }
            s = c+s;
            lena --;
            lenb --;
        }
        while(lena>=0){
        	c = (a[lena]-'0')+ temp +'0';
        	temp = 0;
        	if((c-'0')>=2){
        		temp = 1;
        		c = c-2;
        	}
        	s = c+s;
        	lena--;
        }
        while(lenb>=0){
        	c = (b[lenb]-'0')+ temp +'0';
        	temp = 0;
        	if((c-'0')>=2){
        		temp = 1;
        		c = c-2;
        	}
        	s = c+s;
        	lenb--;
        }
        if(temp == 1){
        	s ='1'+s;
        }
        return s;
    }
};

File: 11二进制求和.py

def addBinary( a, b):
    lena = len(a)
    lenb = len(b)
    flag = False
    l = max(lena, lenb)
    ans = ''
    for i in range(1,l+1):
        t = (0 if i>lena else int(a[-i])) + (0 if i>lenb else int(b[-i])) + (1 if flag else 0 )
        flag = t/2==1
        ans = str(t%2) +ans
    return ('1' if flag else '0') + ans


def main():
    da = addBinary('11100', '101')
    print(da)

if __name__ == '__main__':
    main()
